src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -233,11 +282,72 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
+            return new ValueMappingEstimator<TInputType, TOutputType>(CatalogUtils.GetEnvironment(catalog), keys, values,
+                new[] { (outputColumnName, inputColumnName ?? outputColumnName) });
+        }
+
+        /// <summary>
+        /// <see cref="ValueMappingEstimator"/>
+        /// </summary>
+        /// <typeparam name="TInputType">The key type.</typeparam>
+        /// <typeparam name="TOutputType">The value type.</typeparam>
+        /// <param name="catalog">The conversion transform's catalog</param>
+        /// <param name="keyValuePairs">Specifies the mapping that will be perfomed. The keys will be mapped to the values as specified in the <paramref name="keyValuePairs"/>.</param>
+        /// <param name="columns">The columns to apply this transform on.</param>
+        /// <returns>An instance of the <see cref="ValueMappingEstimator"/></returns>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[ValueMappingEstimator](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/ValueMapping.cs)]
+        ///  [!code-csharp[ValueMappingEstimator](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/ValueMappingStringToKeyType.cs)]
+        ///  [!code-csharp[ValueMappingEstimator](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/ValueMappingFloatToString.cs)]
+        ///  [!code-csharp[ValueMappingEstimator](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/ValueMappingStringToArray.cs)]
+        /// ]]></format>
+        /// </example>
+        [BestFriend]
+        internal static ValueMappingEstimator<TInputType, TOutputType> MapValue<TInputType, TOutputType>(
+            this TransformsCatalog.ConversionTransforms catalog,
+            IEnumerable<KeyValuePair<TInputType, TOutputType[]>> keyValuePairs,
             params ColumnOptions[] columns)
-            => new ValueMappingEstimator<TInputType, TOutputType>(CatalogUtils.GetEnvironment(catalog), keys, values,
-                ColumnOptions.ConvertToValueTuples(columns));
+        {
+            var keys = keyValuePairs.Select(pair => pair.Key);
+            var values = keyValuePairs.Select(pair => pair.Value);
+            return new ValueMappingEstimator<TInputType, TOutputType>(CatalogUtils.GetEnvironment(catalog), keys, values,
+                    ColumnOptions.ConvertToValueTuples(columns));
+        }
+
+        /// <summary>
+        /// <see cref="ValueMappingEstimator"/>
+        /// </summary>
+        /// <param name="catalog">The conversion transform's catalog</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
+        /// <param name="lookupMap">An instance of <see cref="IDataView"/> that contains the key and value columns.</param>
+        /// <param name="keyColumn">The key column in <paramref name="lookupMap"/>.</param>
```

Comment:
>keyColumn [](start = 25, length = 9)

keyColumnName

Reply:
Since I made it `DataViewSchema.Column` I think it makes sense to keep it as is. I changed in the below extension to remove `Name` from the name of `keyColumn` and `valueColumn`.

---
In reply to: [266693603](https://github.com/dotnet/machinelearning/pull/2959#discussion_r266693603) [](ancestors = 266693603)